end-user-application-machines/new-agent-vm/Install-Idle-Shutdown.sh

```diff
@@ -0,0 +1,282 @@
+}
+
+####################################################################
+# If there are active remote ssh connections or we detect PCoIP return 1, else 0
+# Note that a remote ssh session is one where the remote address is NOT "0.0.0.0" (ipv4
+# or "::0" or "::" (ipv6)
+####################################################################
+function activeConnectionsExist() {
+    set -x
+
+    # first, check to see if anyone is connected remotely
+    #  last will produce something like the list below,
+
+    #  someuser  pts/2         tervdiw10dev13.t Wed Jul 12 10:51   still logged in
+    #  someuser  pts/1         0.0.0.0          Wed Jul 12 10:36   gone - no logout
+    #  someuser  pts/0         0.0.0.0          Wed Jul 12 10:35   gone - no logout
+    #  someuser  :100          0.0.0.0          Wed Jul 12 10:35   gone - no logout
+    #  reboot   system reboot  0.0.0.0          Wed Jul 12 10:35   gone - no logout
+    #  
+    #  wtmp begins Wed Jul 12 10:35:17 2017
+
+    # in this case, pts/2 has a remotely logged on user, but the rest are local ssh connections we
+    # can ignore, including the system one.  Format is slightly different when the system status is displayed
+        
+    ipList=$(/usr/bin/last  | /usr/bin/grep 'still logged in' | /usr/bin/awk '{ if (!($0~"wtmp begins")) { if ($2 == "system") {print $4} else {print $3 }}}')
+
+    for i in $ipList; do
+        if [ $i != "0.0.0.0" ] && [ $i != "::" ] && [ $i != "::0" ]; then
+            # a remote ssh connection detected
+            echo 1
+            return
+        fi
+    done
+
+    # Is PCoIP active?
+    /bin/ps -e | /bin/grep -s pcoip-server  > /dev/null 2>&1
+    rc=$?
+
+    if [ $rc -eq 0 ]; then
+        # someone is connected via pcoip
+        echo 1
+        return
+    fi
+
+    echo 0
+    return
+}
+
+####################################################################
+# convert string to integer, 0 if it is invalid
+####################################################################
+function validateInt(){
+    typeset -r input=$1
+    if [[ $input =~ ^[0-9]+$ ]]; then
+        echo $input
+        return 
+    fi
+    echo 0
+}
+
+####################################################################
+# Returns an integer representing the percentage cpuload of the busiest
+# cpu.  e.g. 10 = 10%
+# We do not attempt to normalize
+####################################################################
+function getCPULoad() {
+    # get load average for last 15 minutes
+
+    loadAvg=$(/usr/bin/uptime | sed s/"^.*load average:"// | sed s/","// | /usr/bin/awk '{ print $3}')
+    result=$(echo "($loadAvg * 100 )" | bc)
+    result=${result%%.*}
+    if [ $result -gt 100 ]; then
+        result=100
+    fi
+    echo $result
+}
+
+####################################################################
+# Set the globals wait_minutes and poll_interval
+####################################################################
+function processArgs () {
```

Comment:
I'm not sure I understand why the script accepts these values as command-line args when we are only running it in a non-interactive context. If the interactivity isn't required, we could remove the entire polling loop and simplify the script by using a [systemd.timer unit](http://0pointer.de/public/systemd-man/systemd.timer.html) to periodically execute the corresponding service.